extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1174,11 +1178,27 @@ export class AmpA4A extends AMP.BaseElement {
             'src': srcPath,
             'name': nameData,
           }, SHARED_IFRAME_PROPERTIES));
+      if (method == XORIGIN_MODE.NAMEFRAME) {
+        const _context = this.generateSentinelAndContext(iframe);
+        const name = JSON.stringify({creative, _context});
+        iframe.setAttribute('name', name);
+      }
       return this.iframeRenderHelper_(iframe);
     });
   }
 
   /**
+   * Generates sentinel for iframe and gets the context metadata.
+   * @param {!Element} iframe
+   * @return {!Object} context
+   */
+  generateSentinelAndContext(iframe) {
+    const sentinel = generateSentinel(window);
+    const context = getContextMetadata(window, iframe, sentinel)._context;
```

Comment:
looking at the impl of `getContextMetadata`, it seems weird that we're building an `attributes` object but end up using on one field of it. can't we just extract that part out as a separate method?